src/core/utils/stop-watch.cpp

```diff
@@ -42,12 +42,18 @@ Stopwatch& getProfiler() {
   return s;
 }
 
-Stopwatch::Stopwatch(StopwatchMode _mode) : active(true), mode(_mode) {
+Stopwatch::Stopwatch(StopwatchMode _mode) : active(true), mode(_mode), profiler_active(false) {
   records_of = new map<string, PerformanceData>();
 }
 
 Stopwatch::~Stopwatch() { delete records_of; }
 
+void Stopwatch::enable_profiler() { profiler_active = true; }
+
+void Stopwatch::disable_profiler() { profiler_active = false; }
+
+bool Stopwatch::profiler_status() { return profiler_active; }
+
 void Stopwatch::set_mode(StopwatchMode new_mode) { mode = new_mode; }
 
 bool Stopwatch::performance_exists(string perf_name) { return (records_of->find(perf_name) != records_of->end()); }
```

Comment:
Weird.

Reply:
That piece of the code was included by @andreadelprete. So, he will the best to answer/handle your concern.

Btw, could you tell us what exactly you mean with "weird"?

Reply:
This code is performing a copy, which is weird for a profiler ;)

Reply:
As far I understand, this function is used when we want to "report" *(i.e., print) the results of the profilers. It shouldn't affect the estimation of the computation time.

I will mark this conversation as resolved. Please reopen it if you believe so.

In any case, thanks for pointing out this bit.